P0 bug: Round-trip stress scenario crashes 'Progress watchdog already tracks an active scenario' — introduced by #1954, kills entire round-trip job

## What happened

Stress run [29220954180](https://github.com/thomhurst/Dekaf/actions/runs/29220954180) (main @ 888b4ef1): the **Producer → Consumer Round-Trip (Paired)** job failed ~40s after scenario start, before any measurement:

```
Error: System.InvalidOperationException: Progress watchdog already tracks an active scenario.
##[error]Process completed with exit code 1.
```

Deterministic, not flaky. Consequences:
- Zero round-trip data this run (`results-producer-roundtrip-1brokers` artifact is 203 bytes — empty envelope). The #1940 regression (round-trip 0.87x Confluent) is **unverifiable** post-fix.
- `Publish Stress Test Results to Documentation` and `Fail on Stress Regressions` were both **skipped**, so the 6 trend-gate breaches present in this run's JSON were never evaluated.

## Root cause

`ProducerRoundTripStressTest.cs` registers two overlapping watchdog registrations on the **same** `ProgressWatchdog`:

- Line ~86: `using var watchdog = options.ProgressWatchdog.Track(throughput, Client, Name, ...)` — produce phase, scoped to the whole method.
- Line ~146: `using var consumeWatchdog = options.ProgressWatchdog.Track(consumeProgress, Client, $"{Name}-consume", ...)` — consume phase, while the first registration is still alive.

`ProgressWatchdog.Track` throws when `_activeRun is not null` (`ProgressWatchdog.cs:156`). The second `Track` therefore always throws once the produce loop finishes.

Introduced by #1954 ("Restore round-trip batching and phase diagnostics"), which added the phase-separated consume watchdog (per #1946/#1948) without releasing the produce registration.

`SoakStressTest` already solves this correctly for its concurrent phases: it creates a second watchdog via `options.ProgressWatchdog.CreateSibling()` (`SoakStressTest.cs:119`).

## Fix

Either:
1. Scope the produce watchdog so it is disposed before the consume phase starts (phases are sequential — a nested `using` block around the produce loop + flush suffices), or
2. Use `CreateSibling()` for the consume-phase watchdog, matching `SoakStressTest`.

Option 1 is cleaner since the phases never overlap. Also applies to the Confluent round-trip path if it shares the structure (its single `Track` at line ~326 appears fine, but verify).

## Acceptance

- Round-trip stress job completes and reports phase-separated produce/consume rates.
- Add a unit test that runs a scenario with two sequential phase `Track` registrations against one watchdog to lock in the pattern.

--- a/tests/Dekaf.Tests.Unit/StressTests/ProgressWatchdogTests.cs
+++ b/tests/Dekaf.Tests.Unit/StressTests/ProgressWatchdogTests.cs
@@ -140,6 +140,30 @@ public sealed class ProgressWatchdogTests
         }
     }
 
+    [Test]
+    public void Track_SequentialPhases_ReusesWatchdog()
+    {
+        var outputDirectory = CreateOutputDirectory();
+        try
+        {
+            using var watchdog = new ProgressWatchdog(outputDirectory);
+            var produceProgress = new ThroughputTracker();
+            var consumeProgress = new ThroughputTracker();
+
+            using (watchdog.Track(produceProgress, "Dekaf", "producer-roundtrip"))
+            {
+            }
+
+            using (watchdog.Track(consumeProgress, "Dekaf", "producer-roundtrip-consume"))
+            {
+            }
+        }
+        finally
+        {
+            Directory.Delete(outputDirectory, recursive: true);
+        }
+    }
+
     private static string CreateOutputDirectory()
     {
         var path = Path.Combine(Path.GetTempPath(), $"dekaf-watchdog-tests-{Guid.NewGuid():N}");

--- a/tools/Dekaf.StressTests/Scenarios/ProducerRoundTripStressTest.cs
+++ b/tools/Dekaf.StressTests/Scenarios/ProducerRoundTripStressTest.cs
@@ -83,45 +83,48 @@ internal sealed class ProducerRoundTripStressTest : IStressTestScenario
         await using var sampler = StressTestHelpers.StartSampler(throughput, cancellationToken);
         var produceTimer = Stopwatch.StartNew();
 
-        using var watchdog = options.ProgressWatchdog.Track(
-            throughput,
-            Client,
-            Name,
-            () => StressTestHelpers.CaptureProducerDeliveryDiagnostics(producer, options));
-        using var produceTimeout = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-        produceTimeout.CancelAfter(RoundTripScenarioHelpers.GetTimeout(options));
-
-        Console.WriteLine($"  Producing {options.RoundTripMessages:N0} sequenced messages with Dekaf...");
-        for (var ordinal = 0; ordinal < options.RoundTripMessages; ordinal++)
+        using (options.ProgressWatchdog.Track(
+                   throughput,
+                   Client,
+                   Name,
+                   () => StressTestHelpers.CaptureProducerDeliveryDiagnostics(producer, options)))
+        using (var produceTimeout = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
         {
-            if (RoundTripScenarioHelpers.TryRecordProduceTimeout(
-                    produceTimeout.IsCancellationRequested,
-                    throughput,
-                    client: "Dekaf",
-                    ordinal: ordinal,
-                    cancellationToken: cancellationToken))
+            produceTimeout.CancelAfter(RoundTripScenarioHelpers.GetTimeout(options));
+
+            Console.WriteLine($"  Producing {options.RoundTripMessages:N0} sequenced messages with Dekaf...");
+            for (var ordinal = 0; ordinal < options.RoundTripMessages; ordinal++)
             {
-                break;
+                if (RoundTripScenarioHelpers.TryRecordProduceTimeout(
+                        produceTimeout.IsCancellationRequested,
+                        throughput,
+                        client: "Dekaf",
+                        ordinal: ordinal,
+                        cancellationToken: cancellationToken))
+                {
+                    break;
+                }
+
+                var message = factory.Create(ordinal % options.Partitions);
+                try
+                {
+                    await producer.FireAsync(options.Topic, message.Key, message.Value).ConfigureAwait(false);
+                    throughput.RecordMessage(message.Value.Length);
+                }
+                catch (Exception ex)
+                {
+                    throughput.RecordError(ex, "Round-trip produce", ordinal);
+                }
+
+                if ((ordinal + 1) % 50_000 == 0)
+                {
+                    Console.WriteLine($"  Produced {ordinal + 1:N0} / {options.RoundTripMessages:N0}");
+                }
             }
 
-            var message = factory.Create(ordinal % options.Partitions);
-            try
-            {
-                await producer.FireAsync(options.Topic, message.Key, message.Value).ConfigureAwait(false);
-                throughput.RecordMessage(message.Value.Length);
-            }
-            catch (Exception ex)
-            {
-                throughput.RecordError(ex, "Round-trip produce", ordinal);
-            }
-
-            if ((ordinal + 1) % 50_000 == 0)
-            {
-                Console.WriteLine($"  Produced {ordinal + 1:N0} / {options.RoundTripMessages:N0}");
-            }
+            await StressTestHelpers.FlushWithTimeoutAsync(producer, throughput).ConfigureAwait(false);
         }
 
-        await StressTestHelpers.FlushWithTimeoutAsync(producer, throughput).ConfigureAwait(false);
         produceTimer.Stop();
         var producerDiagnostics = StressTestHelpers.CaptureProducerDeliveryDiagnostics(producer, options);
         await sampler.StopAsync().ConfigureAwait(false);

